### PR TITLE
docs: added Auth with Casbin in Apache APISIX blog

### PR DESCRIPTION
**Pull request by Serendipity96**
Changes:

docs: added Auth with Casbin in Apache APISIX blog

### CODE REVIEW

**Review by KishaniKandasamy — APPROVED** [review]

**Review by bzp2010 — COMMENTED** [review]
Can file `website/static/img/blog_img/2021-08-18-1.png` use JPEG format (.jpg) to reduce size?

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+[Apache APISIX](https://github.com/apache/apisix) 是一个动态、实时、高性能的 API 网关， 提供负载均衡、动态上游、灰度发布、精细化路由、限流限速、服务降级、服务熔断、身份认证、可观测性等数百项功能。你可以使用 Apache APISIX 来处理传统的南北向流量，以及服务间的东西向流量， 也可以当做 [k8s ingress controller](https://github.com/apache/apisix-ingress-controller) 来使用。
+
+### Casbin
+
+[Casbin](https://casbin.org/zh-CN/) 是一个强大的、高效的开源访问控制框架，其权限管理机制支持多种访问控制模型。
+
+### authz-casbin 插件介绍
+
+在 Apache APISIX 的使用中，路由匹配和请求授权之间有个隐含的矛盾点：为了更高细粒度的权限控制，需要配置更高细粒度的路由，来精准识别请求并对请求进行授权。在复杂的授权模型场景下，这将导致路由数量成倍增加，加剧了运维复杂度。
+authz-casbin 是一个基于 lua-casbin  的 Apache APISIX 插件，支持基于各种访问模型的强大授权。Casbin 是一个强大的、高效的开源访问控制框架，支持 ACL、RBAC、ABAC 等访问控制模型，lua-casbin 是 Lua 版本的授权访问模型。
```
Comment (liuxiran) [inline website/blog/Auth-with-Casbin-in-Apache-APISIX.md]:
```suggestion

authz-casbin 是一个基于 lua-casbin  的 Apache APISIX 插件，支持基于各种访问模型的强大授权。Casbin 是一个强大的、高效的开源访问控制框架，支持 ACL、RBAC、ABAC 等访问控制模型，lua-casbin 是 Lua 版本的授权访问模型。
```

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+authz-casbin 插件可以把路由匹配和请求授权这两个功能很好地进行解耦，你可以将各种授权访问模型加载到 Apache APISIX 中，借助 lua-casbin 实现高效复杂的授权模式。
+
+**注意**：如果你想要实现身份验证（authentication），你需要使用其他插件或者自己来配置完成验证用户身份。
```
Comment (liuxiran) [inline website/blog/Auth-with-Casbin-in-Apache-APISIX.md]: how about  recommend available authentication plugins to users and add internal links to the plugins doc here

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md (hunk 1 above)
Comment (bzp2010) [reply website/blog/Auth-with-Casbin-in-Apache-APISIX.md]: > lua-casbin 是 Lua 版本的授权访问模型

At the same time, should this part be changed to `lua-casbin 是casbin访问控制框架的 Lua 版本实现`?

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+模型里的 matcher 表明：
+
+1. `(g(r.sub, p.sub) || keyMatch(r.sub, p.sub))`：请求里的 subject 和策略里的 subject 有着相同的角色或者请求里的 subject 和策略里的 subject 可以通过内置的方法 `keyMatch` 匹配。`keyMatch` 作为 Lua Casbin 的内置函数，相关的描述以及更多的函数可跳转 [lua-casbin](https://github.com/casbin/lua-casbin/blob/master/src/util/BuiltInFunctions.lua)。
+2. `keyMatch(r.obj, p.obj)`：请求里的 object 和策略里的 object 可相互匹配（代指 URL 链接）。
+3. `keyMatch(r.act, p.act)`：请求里的 action 和策略里的 action 可相互匹配（代指 HTTP 请求方法）。
+
+### 在路由上使用插件
+
+一旦你创建了模型和策略，你可以使用 APISIX admin API 在路由上使用。若想使用，你可以模型和策略的文件路径：
```
Comment (liuxiran) [inline website/blog/Auth-with-Casbin-in-Apache-APISIX.md]:
```suggestion
一旦你创建了模型和策略，你可以使用 APISIX Admin API 在路由上使用。若想使用，你可以模型和策略的文件路径：
```

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        }
+    },
+    "upstream": {
+        "nodes": {
+            "127.0.0.1:1980": 1
+        },
+        "type": "roundrobin"
+    },
+    "uri": "/*"
+}
```
Comment (liuxiran) [inline website/blog/Auth-with-Casbin-in-Apache-APISIX.md]:
```suggestion
}'
```

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+```
+
+在这里，username 是传递到 subject 里的用户名。例如，你可以设置"username":"user"来把你定义的`user:alice`传递到 username ，让 username 成为 Alice。
```
Comment (liuxiran) [inline website/blog/Auth-with-Casbin-in-Apache-APISIX.md]:
```suggestion
在这里，username 是传递到 subject 里的用户名。例如，你可以设置`"username":"user"`来把你定义的`"user":"alice"`传递到 username ，让 username 成为 Alice。
```

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
+---
+title: "在 Apache APISIX 中使用 Casbin 进行授权"
+author: 作者：Casbin 社区 & Apache APISIX 社区
+---
```
Comment (moonming) [inline website/blog/Auth-with-Casbin-in-Apache-APISIX.md:13]: no keywords

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+}'
+```
+
+这将会将插件的配置动态地添加到路由中。通过向插件的配置数据中发送更新模型和策略的请求，你可以轻松地更新插件的配置。
+
+## 最后
+
+感谢 Casbin 和 Apache APISIX 社区的开发者们，从 Casbin 社区的开发者 rushitote 提出 issue，到提交 PR，以及 Apache APISIX 社区的开发者积极 review PR，这个跨社区的合作进行地友好而有序，响应 open source makes world better。
+
+来源：https://rushikeshtote.com/blog/apisix-casbin-authorization
```
Comment (moonming) [inline website/blog/Auth-with-Casbin-in-Apache-APISIX.md]: 404 page

**Review by moonming — CHANGES_REQUESTED** [review]

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md (hunk 1 above)
Comment (Serendipity96) [reply website/blog/Auth-with-Casbin-in-Apache-APISIX.md]: I'm not sure,  @tzssangglass please take a look.

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md (hunk 2 above)
Comment (Serendipity96) [reply website/blog/Auth-with-Casbin-in-Apache-APISIX.md]: Good idea. @tzssangglass Can you help me add links ?

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md (hunk 7 above)
Comment (Serendipity96) [reply website/blog/Auth-with-Casbin-in-Apache-APISIX.md]: Let me check.  Thanks.

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+title: "在 Apache APISIX 中使用 Casbin 进行授权"
+author: 作者：Casbin 社区 & Apache APISIX 社区
+keywords:
+- 网关 Casbin
+- APISIX Casbin 
+- Casbin 授权
+- Casbin 插件
+- 访问控制
+- 授权访问模型
+- authz-casbin 使用
```
Comment (moonming) [inline website/blog/Auth-with-Casbin-in-Apache-APISIX.md]:
```suggestion
- Casbin
- APISIX 
- 授权
- API 网关
- 访问控制
- 身份认证
- RBAC
- ALC
```

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+}'
+```
+
+这将会将插件的配置动态地添加到路由中。通过向插件的配置数据中发送更新模型和策略的请求，你可以轻松地更新插件的配置。
+
+## 最后
+
+感谢 Casbin 和 Apache APISIX 社区的开发者们，从 Casbin 社区的开发者 rushitote 提出 issue，到提交 PR，以及 Apache APISIX 社区的开发者积极 review PR，这个跨社区的合作进行地友好而有序，响应 open source makes world better。
+
+来源：https://medium.com/@rushitote/authorization-in-apisix-using-casbin-59b693669d6d
```
Comment (moonming) [inline website/blog/Auth-with-Casbin-in-Apache-APISIX.md]:
```suggestion
来源：[authorization-in-apisix-using-casbin](https://medium.com/@rushitote/authorization-in-apisix-using-casbin-59b693669d6d)
```

**Review by moonming — COMMENTED** [review]
please add English version too

**Comment by Serendipity96** [comment]
> please add English version too

@yzeng25 Let's confirm the schedule of this article English version together.

**Comment by moonming** [comment]
> > please add English version too
> 
> @yzeng25 Let's confirm the schedule of this article English version together.

Isn’t there a link to the English version at the end of the article?

**Comment by yzeng25** [comment]
> > please add English version too
> 
> @yzeng25 Let's confirm the schedule of this article English version together.

Sure, I am looking at sometime next week ( no later than Wednesday) to get it done.

**Comment by Serendipity96** [comment]
> > > please add English version too
> > 
> > 
> > @yzeng25 Let's confirm the schedule of this article English version together.
> 
> Sure, I am looking at sometime next week ( no later than Wednesday) to get it done.

Maybe the link at the end of the article will help you.

**Comment by Serendipity96** [comment]
> > > please add English version too
> > 
> > 
> > @yzeng25 Let's confirm the schedule of this article English version together.
> 
> Isn’t there a link to the English version at the end of the article?

No, the content needs modify, it is the article‘s materials.

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md (hunk 1 above)
Comment (tzssangglass) [reply website/blog/Auth-with-Casbin-in-Apache-APISIX.md]:
```suggestion
authz-casbin 是一个基于 lua-casbin 的 Apache APISIX 插件，支持基于各种访问模型的强大授权。Casbin 是一个强大的、高效的开源访问控制框架，支持 ACL、RBAC、ABAC 等访问控制模型，lua-casbin 是 Casbin 访问控制框架的 Lua 版本实现。
```

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md (hunk 2 above)
Comment (tzssangglass) [reply website/blog/Auth-with-Casbin-in-Apache-APISIX.md]:
```suggestion
**注意**：如果你想要实现身份验证（authentication），你需要使用其他插件或者自己来配置完成验证用户身份，比如 [jwt-auth](https://github.com/apache/apisix/blob/master/docs/zh/latest/plugins/jwt-auth.md) 插件。
```

File: website/blog/Auth-with-Casbin-in-Apache-APISIX.md [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+    },
+    "uri": "/*"
+}'
+```
+
+这将会将插件的配置动态地添加到路由中。通过向插件的配置数据中发送更新模型和策略的请求，你可以轻松地更新插件的配置。
+
+## 最后
+
+感谢 Casbin 和 Apache APISIX 社区的开发者们，从 Casbin 社区的开发者 rushitote 提出 issue，到提交 PR，以及 Apache APISIX 社区的开发者积极 review PR，这个跨社区的合作进行地友好而有序，响应 open source makes world better。
```
Comment (tzssangglass) [inline website/blog/Auth-with-Casbin-in-Apache-APISIX.md]:
```suggestion
感谢 Casbin 和 Apache APISIX 社区的开发者们，从 Casbin 社区的开发者 rushitote 提出 issue，提交 PR，到 Apache APISIX 社区的开发者们积极地 review PR，这个跨社区的合作友好而有序地向前推进，响应 open source makes world better。
```

**Review by moonming — APPROVED** [review]

**Comment by liuxiran** [comment]
cc @moonming to have a look

**Review by liuxiran — APPROVED** [review]

**Comment by yzeng25** [comment]
English version of this blog is added in this pull request https://github.com/apache/apisix-website/pull/510, it should be available by the end of this week.

**Pull request merged**